### ISSUE TITLE
Google Workspace IdP Sync Requires Manual Restart of Management Container

### ISSUE THREAD

**Issue by ginsul** [netbirdio/netbird#3003]
**Describe the problem**

The Google Workspace IdP synchronization is not functioning as expected, requiring a manual restart of the management container using docker compose restart management to update the IdP settings. 

```
docker compose  restart management
```

**To Reproduce**

1. Go to 'google workspace admin'
2. Rename User
3. After wait 10minutes more, on netbird dashboard still no changes

**Expected behavior**

Sync between netbird dashboard and google workspace

**NetBird version**
netbird selfhosted 0.34.1

**Screenshots**

On Netbird Dashboard : 
![image](https://github.com/user-attachments/assets/3867c144-d5ae-4cbb-8177-b9336a05d6e4)

On Google Workspace :
![image](https://github.com/user-attachments/assets/e8cf46ba-356d-44b7-90e0-3e3169cf32dc)

**Additional context**

Should the management service require a manual restart for IdP sync?
Should user deletions in Google Workspace automatically reflect in Netbird?

**Comment by mlsmaycon** [comment]
@ginsul, the user name and emails are refreshed either on every restart or when a new user logs in to the system.

**Comment by ginsul** [comment]
Thank you for your reply. How about when a user is deleted from Google
Workspace? Is the user automatically removed from NetBird as well? (I
haven’t tried it myself yet.)

**Comment by mlsmaycon** [comment]
No, it is not removed. Currently, that might create a problem on startup since both should be in sync, management store, and Google IDP. 

We plan to fix that soon, but in the mean time, is best to work on an API check script for the API, so that you can delete it when this happens.

**Comment by nazarewk** [comment]
Hello @ginsul,

We're currently reviewing our open issues and would like to verify if this problem still exists in the [latest NetBird version](https://github.com/netbirdio/netbird/releases).

Could you please confirm if the issue is still there?

We may close this issue temporarily if we don't hear back from you within **2 weeks**, but feel free to reopen it with updated information.

Thanks for your contribution to improving the project!

**Comment by mlsmaycon** [comment]
closing issue due to no recent feedback. Feel free to open a new one if the issue persist or reopen if this was a feature request.
